agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -100,6 +101,35 @@ namespace beerocks {
 #define FSM_IS_IN_STATE(eState) (m_eFSMState == EState::eState)
 #define FSM_CURR_STATE_STR s_arrStates[int(m_eFSMState)]
 
+//////////////////////////////////////////////////////////////////////////////
+/////////////////////////////// Helper Functions /////////////////////////////
+//////////////////////////////////////////////////////////////////////////////
+
+template <class SlavesContainer, class MAC>
```

Comment:
The original thought was that we have mac as a struct and mac as a string, so by specializing the template with string it would be transparent to the user. This thought lead to another thought, since it is a helper function, loosely coupled to the backhaul (the return type is a backhaul inner type) why not free the container to be whatever it is, as long as it is forward-iterable.

Reply:
I don't like this either, for either template argument.
- MAC should always be sMacAddr. If there's a string anywhere, call mac_from_string or (better) refactor so that it's not a string.
- Using a template to work around private is super-ugly IMO. I see two better ways to handle this:
    1. Make it a private function of backhaul_manager. Yes, I agree that the class is already overloaded, but still...
    2. Make sRadioInfo a public class (which would imply renaming it to RadioInfo, and moving it to a separate file). This opens opportunities to add more members to RadioInfo, which is what you wanted I think. While you're at it, you'd probably want to add a class RadioList or something like that that abstracts the container.

I leave it up to you to choose which one is better.